src/hotspot/share/gc/shared/gcLocker.cpp

```diff
@@ -123,33 +106,30 @@ bool GCLocker::should_discard(GCCause::Cause cause, uint total_collections) {
          (_total_collections != total_collections);
 }
 
-void GCLocker::jni_lock(JavaThread* thread) {
+void GCLocker::stall_until_no_needs_gc(JavaThread* thread) {
```

Comment:
The increment of _jni_lock_count seems to have gotten lost.  I don't see how
this change can work without that.

Reply:
The `jni_lock` and `jni_unlock` naming and the matching increment/decrement gives one the impression that one is undoing the effect of the other. However, I believe that impression is incorrect. The correct value of `_jni_lock_count` is set by `set_jni_lock_count` inside a safepoint. IOW, the increment inside `jni_lock` will always be overwritten.